src/core/mormot.core.zip.pas

```diff
@@ -2813,7 +2813,7 @@ function TZipRead.UnZip(aIndex: integer; const DestDir: TFileName;
     if not SafeFileName(LocalZipName) then
       raise ESynZip.CreateUtf8('%.UnZip(%): unsafe file name ''%''',
         [self, fFileName, LocalZipName]);
-    Dest := EnsureDirectoryExists(DestDir + ExtractFilePath(LocalZipName));
+    Dest := EnsureDirectoryExists(MakePath([DestDir, ExtractFilePath(LocalZipName)]));

```

Comment:
My guess is that the correct fix is
`Dest := EnsureDirectoryExists(EnsureDirectoryExists(DestDir) + ExtractFilePath(LocalZipName));`